light/errors.go

```diff
@@ -5,9 +5,43 @@ import (
 	"fmt"
 	"time"
 
+	cmtbytes "github.com/cometbft/cometbft/libs/bytes"
+	cmtmath "github.com/cometbft/cometbft/libs/math"
+	"github.com/cometbft/cometbft/light/provider"
 	"github.com/cometbft/cometbft/types"
 )
 
+var (
+
+	// ErrFailedHeaderCrossReferencing is returned when the detector was not able to cross reference the header
+	// with any of the connected witnesses.
+	ErrFailedHeaderCrossReferencing = errors.New("all witnesses have either not responded, don't have the " +
+		" blocks or sent invalid blocks. You should look to change your witnesses" +
+		"  or review the light client's logs for more information")
+	// ErrLightClientAttack is returned when the light client has detected an attempt
+	// to verify a false header and has sent the evidence to either a witness or primary.
+	ErrLightClientAttack = errors.New(`attempted attack detected.
+Light client received valid conflicting header from witness.
+Unable to verify header. Evidence has been sent to both providers.
+Check logs for full evidence and trace`,
+	)
+
+	// ErrNoWitnesses means that there are not enough witnesses connected to
+	// continue running the light client.
+	ErrNoWitnesses               = errors.New("no witnesses connected. please reset light client")
+	ErrNilOrSinglePrimaryTrace   = errors.New("nil or single block primary trace")
+	ErrHeaderHeightAdjacent      = errors.New("headers must be non adjacent in height")
+	ErrHeaderHeightNotAdjacent   = errors.New("headers must be adjacent in height")
+	ErrNegativeOrZeroPeriod      = errors.New("negative or zero period")
+	ErrNegativeHeight            = errors.New("negative height")
+	ErrZeroOrNegativeHeight      = errors.New("zero or negative height")
```

Comment:
> aren't `ErrZeroOrNegativeHeight` and `ErrNegativeOrZeroHeight` the same?

Their message is not the same🤣. I felt bad during the export of these errors... As there are many similar errors with different error messages, we couldn't change them to the same. I'm afraid if we change it, it may affect others who are dependent on the error message😮‍💨

Reply:
Yeah, but until we've released v1, we shouldn't care much about breaking stuff. I think in this case, it's okay to merge these two errors

Reply:
Okay. 

BTW, as this is a huge and long-term project to export all package errors, it would be better if there was a simple guide for exporting the errors so that successors who are going to pick up other packages could start their coding easily.

Such as telling developers if they could adjust the error message, or how to export the errors in some common cases, etc.